workflow/Snakefile

```diff
@@ -9,85 +9,84 @@ import yaml
+make_biosample_config(
+	config["cell_clusters"],
+	os.path.join(
+		config["results_dir"], 
+		"tmp", 
+		"config_abc_biosamples.tsv"
+	),
+	config["results_dir"]
+)
+config["ABC_BIOSAMPLES"] = os.path.join(
+	config["results_dir"], 
+	"tmp", 
+	"config_abc_biosamples.tsv"
+	)
+
+
+## Import ENCODE_rE2G pipeline as a reusable module 
 module encode_e2g:
 	snakefile:
-		f"{E2G_DIR_PATH}/workflow/Snakefile"
-	config: get_e2g_config(config)
-
-use rule * from encode_e2g exclude add_external_features
-
-rule add_external_features:
-	input:
-		predictions_extended = os.path.join(RESULTS_DIR, "{biosample}", "ActivityOnly_features.tsv.gz"),
-		arc_predictions = os.path.join(RESULTS_DIR, "{biosample}", "EnhancerPredictionsAllPutative_ARC.tsv.gz"),
-		feature_table_file = lambda wildcards: get_feature_table_file(wildcards.biosample)
-	output:
-		plus_external_features = os.path.join(RESULTS_DIR, "{biosample}", "ActivityOnly_plus_external_features.tsv.gz")
-	conda:
-		"envs/sc_e2g.yml"
-	script:
-		"scripts/integrate_kendall.R"
-
-rule kendall_correlation:
-	input:
-		kendall_candidate_e2g_pairs = lambda wildcards: BIOSAMPLE_DF.loc[wildcards.biosample, 'candidate_pairs'],
-		sc_atac_matix = lambda wildcards: BIOSAMPLE_DF.loc[wildcards.biosample, 'ATAC_matrix'],
-		sc_rna_matix = lambda wildcards: BIOSAMPLE_DF.loc[wildcards.biosample, 'RNA_matrix']
-	output:
-		kendall_predictions = os.path.join(RESULTS_DIR, "{biosample}", "EnhancerPredictionsAllPutative_Kendall.tsv.gz") 
-	conda:
-		"envs/sc_e2g.yml"
-	script:
-		"scripts/compute_kendall.R"
-
-rule arc_e2g:
-	input:
-		abc_predictions = os.path.join(RESULTS_DIR, "{biosample}", "Predictions", "EnhancerPredictionsAllPutative.tsv.gz"),
-		kendall_predictions = os.path.join(RESULTS_DIR, "{biosample}", "EnhancerPredictionsAllPutative_Kendall.tsv.gz")
-	output:
-		arc_predictions = os.path.join(RESULTS_DIR, "{biosample}", "EnhancerPredictionsAllPutative_ARC.tsv.gz")
-	conda:
-		"envs/sc_e2g.yml"
-	script:
-		"scripts/compute_arc_e2g.R"
-
-	
\ No newline at end of file
+		os.path.join(config["encode_re2g_dir"], "workflow", "Snakefile")
+	config: get_e2g_config(config, config["encode_re2g_dir"])
+
+
+## Include Snakemake rules from ENCODE_rE2G
+use rule * from encode_e2g exclude add_external_features, all
+
+
+## Include additional Snakemake rules
+include: os.path.join("rules", "generate_frag_file.smk")
+include: os.path.join("rules", "frag_to_tagAlign.smk")
+include: os.path.join("rules", "make_kendall_pairs.smk")
+include: os.path.join("rules", "generate_atac_matrix.smk")
+include: os.path.join("rules", "compute_kendall.smk")
+include: os.path.join("rules", "arc_e2g.smk")
+include: os.path.join("rules", "add_external_features.smk")
+
+
+## Import cell cluster configuration table and make a list of cell clusters
+CELL_CLUSTER_DF = pd.read_table(config["cell_clusters"]).set_index("cluster", drop=False)
+CLUSTER = CELL_CLUSTER_DF["cluster"].to_list()
+
+
+## Define the final outputs of the pipeline
+rule all:
```

Comment:
Curious why we needed to redefine this. Since we passed in ABC_BIOSAMPLES to the encode_r2eg module, I'd imagine everything should've still worked.